src/go/rpk/pkg/cli/cluster/connections/connections.go

```diff
@@ -0,0 +1,224 @@
+}
+
+type flagFilters struct {
+	state                       string
+	ipAddress                   string
+	clientID                    string
+	clientSoftwareName          string
+	clientSoftwareVersion       string
+	groupID                     string
+	idleMs                      int64
+	authenticationUserPrincipal string
+}
+
+func (f *flagFilters) buildFilterClauses() ([]string, error) {
+	clauses := []string{}
+
+	if f.state != "" {
+		val, ok := stateOptionMap[f.state]
+		if !ok {
+			return nil, fmt.Errorf("%w: %s", errInvalidStateFilter, f.state)
+		}
+		clauses = append(clauses, fmt.Sprintf("state = %s", val.String()))
+	}
+
+	// We have a series of string options we need to map to the upstream API filters
+	for key, val := range map[string]string{
+		"source.ip_address":                  f.ipAddress,
+		"client_id":                          f.clientID,
+		"client_software_name":               f.clientSoftwareName,
+		"client_software_version":            f.clientSoftwareVersion,
+		"group_id":                           f.groupID,
+		"authentication_info.user_principal": f.authenticationUserPrincipal,
+	} {
+		if val != "" {
+			clauses = append(clauses, fmt.Sprintf("%s = %q", key, val))
+		}
+	}
+
+	if f.idleMs > 0 {
+		clauses = append(clauses, fmt.Sprintf("idle_duration > %dms", f.idleMs))
+	}
+	return clauses, nil
+}
+
+func newConnectionList(fs afero.Fs, p *config.Params) *cobra.Command {
+	var filters flagFilters
+	var orderBy string
+	var limit int32
+
+	cmd := &cobra.Command{
+		Use:   "list",
+		Short: "Display statistics about current connections",
+		Long: `Display statistics about current connections.
+
+This command displays a table of active and recently closed connections within the cluster.
+
+Use the --order-by flag to sort the connections by specific values.`,
+		Example: `
+List connections ordered by the time that they've been idle:
+	rpk cluster connections list --order-by="idle_duration desc"
+
+List connections ordered by those that have made the least requests:
+	rpk cluster connections list --order-by="total_request_statistics.request_count asc"
+
+List extended output for open connections in json format:
+	rpk cluster connections list --format=json --state="OPEN"`,
+		Args: cobra.ExactArgs(0),
+		Run: func(cmd *cobra.Command, _ []string) {
+			prof, err := p.LoadVirtualProfile(fs)
+			out.MaybeDie(err, "rpk unable to load config: %v", err)
+
+			// This doesn't support using the public API yet
+			config.CheckExitCloudAdmin(prof)
+
+			cl, err := adminapi.NewClient(cmd.Context(), fs, prof)
+			out.MaybeDie(err, "unable to initialize admin client: %v", err)
+
+			// Build the filters based on the current flag set
+			filterClauses, err := filters.buildFilterClauses()
+			out.MaybeDie(err, "invalid filters: %v", err)
+
+			req := adminv2.ListKafkaConnectionsRequest{
+				Filter:   strings.Join(filterClauses, " AND "),
+				OrderBy:  orderBy,
+				PageSize: limit,
+			}
+
+			svc := cl.BrokerService()
```

Comment:
Is there a reason to assign the service to a single variable instead of chaining it all together?